### ISSUE TITLE
[BUG] Caching does not work with useTable due to the returned loading attribute set to true during re-validation

### ISSUE THREAD

**Issue by IgnusG** [pankod/refine#1321]
**Describe the bug**
With the changes introduced in https://github.com/pankod/refine/pull/1319 `useTable` properly returns loading as true while loading is being performed on new requests but now also incorrectly returns true while react query revalidates the response.

**To Reproduce**
1. Go to list page
2. Click on a new page through pagination
3. Observe that loader correctly displays
4. Click on previous page
5. Observe that loader appears again even though data is already there (react query merely revalidates the cache at this point)

**Expected behavior**
Loading should not appear when a cached response is ready to be served. This can be achieved by relying either on `status` or on `isSuccess` instead of `isFetching` (which is also true while revalidating cache).

See for solution suggestion https://github.com/pankod/refine/pull/1319#issuecomment-993494500

**Screenshots**
Also shown in suggestion mentioned above.

**Desktop (please complete the following information):**
 - Browser: Edge
 - Version: 2.4.5
 - Used dataProvider: Custom Mock Implementation

**Additional context**
I wasn't sure if I should create a new issue for this but thought it's better to keep track of it separately since it deals with something slightly different from https://github.com/pankod/refine/issues/1316.

**Comment by omeraplak** [comment]
Hey @IgnusG,
We tried, but we could not achieve the desired result. Can you take a look too?
![table-loading-min](https://user-images.githubusercontent.com/1110414/146543344-3a292632-a2bb-473d-b988-018b087e55b6.gif)

**Comment by IgnusG** [comment]
Ahh I see. I was wrong with `isSuccess`, sorry for that. The problem is `react-query` returns previous data under `isSuccess` with the flag `isPreviousData` set to true which is why this "fix" broke the loader again.

I've dug around some more and I think the most correct solution is simply using a negated [`isFetched`](https://react-query.tanstack.com/reference/useQuery):

``` typescript
        tableProps: {
            ...tablePropsSunflower,
            dataSource: data?.data,
            loading: liveMode === "auto" ? isLoading : !isFetched,
        ...
```

This attribute is false if we are loading new data (so the loading becomes true and the loader is shown), but is set to true if the cached data is already available (while `react-query` performs a background refetch).

![refine-react-query-suggestion-2](https://user-images.githubusercontent.com/6438760/146556235-235d2559-912e-4719-94e3-f8fa6246ed9b.gif)

**Comment by omeraplak** [comment]
This looks great. Congrats! 
I'll send a PR about it on Monday. If you want, you can send it too!

**Comment by omeraplak** [comment]
Hey @IgnusG ,
We've released 2.4.11! Thank you
